src/test/scala/com/microsoft/hyperspace/index/rules/RuleUtilsTest.scala

```diff
@@ -84,15 +85,16 @@ class RuleUtilsTest extends HyperspaceRuleTestSuite {
   }
 
   test("Verify indexes are matched by signature correctly.") {
-    val indexManager = IndexCollectionManager(spark)
-
-    assert(RuleUtils.getCandidateIndexes(indexManager, t1ProjectNode).length === 3)
-    assert(RuleUtils.getCandidateIndexes(indexManager, t2ProjectNode).length === 2)
+    assert(RuleUtils.getCandidateIndexes(spark, t1ProjectNode).length === 3)
+    assert(RuleUtils.getCandidateIndexes(spark, t2ProjectNode).length === 2)
 
     // Delete an index for t1ProjectNode
+    val indexManager = Hyperspace
+      .getContext(spark)
+      .indexCollectionManager
```

Comment:
Can you explain why `val indexManager = IndexCollectionManager(spark)` wouldn't work? This is subtle and I expect both to work if they are using the same `spark`.

Reply:
With `val indexManager = IndexCollectionManager(spark)`, the change of below `delete` was not seen inside `getCandidateIndexes`.

Reply:
Do you know what caused this behavior change?

Reply:
I guess it's because `IndexCollectionManager(spark)` creates a new instance?
```
object IndexCollectionManager {
  def apply(spark: SparkSession): IndexCollectionManager =
    new IndexCollectionManager(
      spark,
      IndexLogManagerFactoryImpl,
      IndexDataManagerFactoryImpl,
      FileSystemFactoryImpl)
```

Reply:
Looks like you are calling `val indexManager = Hyperspace.getContext(spark).indexCollectionManager` inside `getCandidateIndexes`. Can you make `getCandidateIndexes` to take in `IndexManager` so that it doesn't have to depend on a specific instance?

Reply:
It makes sense. Reverted the change. Thanks!